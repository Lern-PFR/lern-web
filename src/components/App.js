function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Edit <code>src/components/App.js</code> and save to reload.
        </p>
      </header>
    </div>
  );
}

export default App;
