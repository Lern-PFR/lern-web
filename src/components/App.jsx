import { Home } from 'react-feather';
import { MainLayout, StyledDiv } from 'components/shared/layout';
import {
	Canon,
	Trafalgar,
	Paragon,
	GreatPrimer,
	DoublePica,
	BodyCopy,
	Pica,
	LongPrimer,
	Brevier,
	Minion,
} from 'components/shared/typography';
import {
	PrimaryButton,
	StandardButton,
	SubtleButton,
	DropdownButton,
	DangerButton,
	IconButton,
	FloatingActionButton,
} from 'components/shared/buttons';
import {
	elevationFlat,
	elevationFirst,
	elevationSecond,
	elevationThird,
	elevationFourth,
	elevationFifth,
} from 'theme/elevations';
import {
	LabeledCheckbox,
	FieldsetComponent,
	LabeledRadioButton,
	LabelComponent,
	LabeledInput,
} from 'components/shared/form';

/**
 * @name App
 * @description The main component of the application.
 *
 * @author TimothÃ©e Simon-Franza
 */
const App = () => (
	<MainLayout>
		<StyledDiv position="relative">
			<Canon>Canon text element.</Canon>
			<Trafalgar>Trafalgar text element.</Trafalgar>
			<Paragon>Paragon text element.</Paragon>
			<GreatPrimer>GreatPrimer text element.</GreatPrimer>
			<DoublePica>DoublePica text element.</DoublePica>
			<BodyCopy>BodyCopy text element.</BodyCopy>
			<Pica>Pica text element.</Pica>
			<LongPrimer>LongPrimer text element.</LongPrimer>
			<Brevier>Brevier text element.</Brevier>
			<Minion>Minion text element.</Minion>
		</StyledDiv>
		<StyledDiv position="relative">
			<PrimaryButton>Primary button</PrimaryButton>
			<StandardButton>Standard button</StandardButton>
			<SubtleButton>Subtle button</SubtleButton>
			<DropdownButton>Dropdown button</DropdownButton>
			<DangerButton>Danger button</DangerButton>
			<IconButton><Home /></IconButton>
			<FloatingActionButton isIconOnly><Home /></FloatingActionButton>
		</StyledDiv>
		<StyledDiv position="relative" marginTop="50px" display="flex" width="100%">
			<StyledDiv {...elevationFlat} width="260px" height="260px" padding="10px">Flat component</StyledDiv>
			<StyledDiv {...elevationFirst} width="260px" height="260px" padding="10px">5% elevation component</StyledDiv>
			<StyledDiv {...elevationSecond} width="260px" height="260px" padding="10px">10% elevation component</StyledDiv>
			<StyledDiv {...elevationThird} width="260px" height="260px" padding="10px">20% elevation component</StyledDiv>
			<StyledDiv {...elevationFourth} width="260px" height="260px" padding="10px">30% elevation component</StyledDiv>
			<StyledDiv {...elevationFifth} width="260px" height="260px" padding="10px">50% elevation component</StyledDiv>
		</StyledDiv>
		<StyledDiv position="relative" display="flex" flexDirection="column">
			<LabelComponent>Default label component</LabelComponent>
			<LabelComponent textStyle="bodycopy">BodyCopy label component</LabelComponent>
		</StyledDiv>
		<StyledDiv position="relative">
			<FieldsetComponent title="Basic checkbox components">
				<LabeledCheckbox id="lcb5">Labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox labelTextStyle="bodycopy" checked id="lcb6">Default checked labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox disabled id="lcb7">Disabled labeled checkbox</LabeledCheckbox>
				<LabeledCheckbox disabled checked id="lcb8">Disabled default checked labeled checkbox</LabeledCheckbox>
			</FieldsetComponent>
		</StyledDiv>
		<StyledDiv position="relative">
			<FieldsetComponent title="Basic radio button components">
				<LabeledRadioButton name="rg2" id="lrb5">Labeled radio button</LabeledRadioButton>
				<LabeledRadioButton labelTextStyle="bodycopy" name="rg2" checked id="lrb6">Default checked labeled radio button</LabeledRadioButton>
				<LabeledRadioButton name="rg2" disabled id="lrb7">Disabled labeled radio button</LabeledRadioButton>
			</FieldsetComponent>
		</StyledDiv>
		<StyledDiv position="relative" width="500px">
			<FieldsetComponent title="Text input components">
				<LabeledInput id="basicTextInput">Labeled text input</LabeledInput>
				<LabeledInput id="placeholderTextInput" placeholder="Enter details here.">Labeled text input with custom placeholder</LabeledInput>
				<LabeledInput id="hintTextInput" placeholder="Enter details here." hintText="This is a hint for the user.">Labeled text input with hint</LabeledInput>
				<LabeledInput
					id="disabledTextInput"
					placeholder="Enter details here."
					hintText="This is a hint for the user."
					disabled
				>
					Disabled labeled text input
				</LabeledInput>
				<LabeledInput
					id="errorTextInput"
					placeholder="Enter details here."
					hintText="This is a hint for the user."
					errorText="There is something wrong with the value in this input"
					error
				>
					Labeled text input with a validation error
				</LabeledInput>
				<LabeledInput id="numberInput" placeholder="Number Input" type="number">Labeled number input</LabeledInput>
				<LabeledInput id="passwordInput" placeholder="Password Input" type="password">Labeled password input</LabeledInput>
			</FieldsetComponent>
		</StyledDiv>
	</MainLayout>
);

export default App;
