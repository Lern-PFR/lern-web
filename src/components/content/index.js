import ConceptCreationForm from './ConceptCreationForm';
import ConceptEditionForm from './ConceptEditionForm';
import ConceptLessonList from './ConceptLessonList';
import ModuleCreationForm from './ModuleCreationForm';
import ModuleEditionForm from './ModuleEditionForm';
import ModuleConceptList from './ModuleConceptList';
import NavigationSidebar from './NavigationSidebar';
import SubjectCreationForm from './SubjectCreationForm';
import SubjectEditionForm from './SubjectEditionForm';
import SubjectModuleList from './SubjectModuleList';

export {
	ConceptCreationForm,
	ConceptEditionForm,
	ConceptLessonList,
	ModuleCreationForm,
	ModuleEditionForm,
	ModuleConceptList,
	NavigationSidebar,
	SubjectCreationForm,
	SubjectEditionForm,
	SubjectModuleList,
};
