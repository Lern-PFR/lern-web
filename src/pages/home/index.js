import Homepage from './Homepage';

export {
	// eslint-disable-next-line import/prefer-default-export
	Homepage,
};
