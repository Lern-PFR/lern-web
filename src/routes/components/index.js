import RenderRoutes from './RenderRoutes';

export {
	// TODO: disable this exception once there is more to export
	// eslint-disable-next-line import/prefer-default-export
	RenderRoutes,
};
