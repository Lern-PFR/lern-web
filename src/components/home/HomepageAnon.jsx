import { PrimaryButton } from 'components/shared/buttons';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { StyledDiv, StyledSvg } from 'components/shared/styledElements';
import { BodyCopy, Canon } from 'components/shared/typography';
import {
	layout,
	hero,
	subtitle,
	illustrationSvg,
} from 'theme/pages/home/homepageAnon';
import { Link } from 'components/shared/navigation';
import conf from 'conf';
import routes from 'routes';

/**
 * @name HomepageAnon
 * @description The homepage to display to anonymous (non-connected) users
 *
 * @author Christopher Walker
 *
 * @param {func} t	The translation method provided by the withTranslation HoC.
 */
const HomepageAnon = ({ t }) => (
	<StyledDiv {...layout}>
		<StyledDiv>
			<Canon {...hero} tag="h1">{t('home.pages.anon.hero')}</Canon>
			<BodyCopy {...subtitle}>{t('home.pages.anon.subtitle')}</BodyCopy>
			<PrimaryButton>
				<Link to={routes.subjects.subjectCreation}>{t('home.pages.anon.links.discover')}</Link>
			</PrimaryButton>
		</StyledDiv>
		<StyledSvg src={`${conf.svgIllustrationsPath}/homepage_anon.svg`} {...illustrationSvg} />
	</StyledDiv>
);

HomepageAnon.propTypes = {
	t: PropTypes.func.isRequired,
};

export default withTranslation()(HomepageAnon);
