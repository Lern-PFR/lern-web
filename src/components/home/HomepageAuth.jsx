import { useEffect } from 'react';
import { useDispatch } from 'react-redux';

import { clearSubjectList, fetchSubjectList } from 'redux/actions/subjects';
import { clearProgressionList, fetchProgressionList } from 'redux/actions/progression';

import { PrimaryButton, SubtleButton } from 'components/shared/buttons';
import PropTypes from 'prop-types';
import { withTranslation } from 'react-i18next';
import { StyledDiv } from 'components/shared/styledElements';
import { BodyCopy, Canon } from 'components/shared/typography';
import {
	layout,
	hero,
	subtitle,
	mainContainer,
	lernButton,
} from 'theme/pages/home/homepageAuth';
import { Link } from 'components/shared/navigation';
import routes from 'routes';
import LatestCourses from './LatestCourses';
import Sidebar from './Sidebar';

/**
 * @name HomepageAuth
 * @description The homepage to display to anonymous (non-connected) users
 *
 * @author Christopher Walker
 *
 * @param {func} t	The translation method provided by the withTranslation HoC.
 */
const HomepageAuth = ({ t, user }) => {
	const dispatch = useDispatch();

	useEffect(() => {
		dispatch(fetchSubjectList());
		dispatch(fetchProgressionList());

		return () => {
			dispatch(clearSubjectList());
			dispatch(clearProgressionList());
		};
	}, [dispatch]);

	return (
		<StyledDiv {...layout}>
			<StyledDiv display="flex" flexDirection="column" {...mainContainer}>
				<StyledDiv>
					<Canon {...hero} tag="h1">{t('home.pages.auth.hero', { user })}</Canon>
					<BodyCopy {...subtitle}>{t('home.pages.auth.subtitle')}</BodyCopy>
					<PrimaryButton>
						<Link to={routes.subjects.default}>{t('home.pages.auth.links.subjects')}</Link>
					</PrimaryButton>
					<SubtleButton {...lernButton}>
						<Link to={routes.subjects.subjectStructure}>{t('home.pages.auth.links.structure')}</Link>
					</SubtleButton>
				</StyledDiv>
				<LatestCourses />
			</StyledDiv>
			<Sidebar />
		</StyledDiv>
	);
};

HomepageAuth.propTypes = {
	t: PropTypes.func.isRequired,
	user: PropTypes.object.isRequired,
};

export default withTranslation()(HomepageAuth);
