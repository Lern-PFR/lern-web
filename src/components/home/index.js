import HomepageAnon from './HomepageAnon';
import HomepageAuth from './HomepageAuth';

export {
	HomepageAnon,
	HomepageAuth,
};
