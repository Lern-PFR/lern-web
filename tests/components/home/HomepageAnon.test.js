import { shallow } from 'enzyme';
import { HomepageAnon } from 'components/home';

describe('HomepageAnon', () => {
	it('should match previous snapshot', () => {
		const wrapper = shallow(<HomepageAnon />);

		expect(wrapper).toMatchSnapshot();
	});
});
