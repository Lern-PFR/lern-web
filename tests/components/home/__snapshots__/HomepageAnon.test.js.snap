// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomepageAnon should match previous snapshot 1`] = `
<styled.div
  alignItems="center"
  display="grid"
  gridGap="3em"
  gridTemplateColumns="minmax(700px, 1fr) 1fr"
  marginX="auto"
  maxWidth="1450px"
>
  <styled.div>
    <Canon
      lineHeight="initial"
      tag="h1"
    >
      home.pages.anon.hero
    </Canon>
    <BodyCopy
      lineHeight="initial"
    >
      home.pages.anon.subtitle
    </BodyCopy>
    <PrimaryButton>
      <Link
        to="/subjects/structure"
      >
        home.pages.anon.links.discover
      </Link>
    </PrimaryButton>
  </styled.div>
  <Styled(ReactSVG)
    display="inline-block"
    src="/assets/svg/illustrations/homepage_anon.svg"
  />
</styled.div>
`;
